crates/sui-node/src/lib.rs

```diff
@@ -1034,10 +1034,38 @@ impl SuiNode {
         registry_service: &RegistryService,
         sui_node_metrics: Arc<SuiNodeMetrics>,
     ) -> Result<ValidatorComponents> {
-        let consensus_config = config
-            .consensus_config()
+        let mut config_clone = config.clone();
+        let consensus_config = config_clone
+            .consensus_config
+            .as_mut()
             .ok_or_else(|| anyhow!("Validator is missing consensus config"))?;
 
+        // TODO (mysticeti): Move this to a protocol config flag.
+        let consensus_protocol = if std::env::var("CONSENSUS").is_ok() {
+            let consensus_choice = std::env::var("CONSENSUS").unwrap();
+            match consensus_choice.as_str() {
+                "narwhal" => ConsensusProtocol::Narwhal,
+                "mysticeti" => ConsensusProtocol::Mysticeti,
+                "swap_each_epoch" => {
+                    if epoch_store.epoch() % 2 == 0 {
+                        ConsensusProtocol::Narwhal
+                    } else {
+                        ConsensusProtocol::Mysticeti
+                    }
+                }
+                _ => {
+                    let consensus = consensus_config.protocol.clone();
+                    warn!("Consensus env var was set to an invalid choice, using default consensus protocol {consensus:?}");
+                    consensus
+                }
+            }
+        } else {
+            consensus_config.protocol.clone()
+        };
```

Comment:
nit: could we modify the `consensus_config.protocol` only when the `CONSENSUS` env variable is set?  so basically something like:

```
     if let Ok(consensus_choice) = std::env::var("CONSENSUS") {
            let consensus_protocol = match consensus_choice.as_str() {
                "narwhal" => ConsensusProtocol::Narwhal,
                "mysticeti" => ConsensusProtocol::Mysticeti,
                "swap_each_epoch" => {
                    if epoch_store.epoch() % 2 == 0 {
                        ConsensusProtocol::Narwhal
                    } else {
                        ConsensusProtocol::Mysticeti
                    }
                }
                _ => {
                    let consensus = consensus_config.protocol.clone();
                    warn!("Consensus env var was set to an invalid choice, using default consensus protocol {consensus:?}");
                    consensus
                }
           }     
          consensus_config.protocol = consensus_protocol;
    }
```

it would make me feel a little safer that there is no way to touch the original setting unless the env variable has been explicitly set.